File: datasets/JPlag-Java/to-kotlin-intellij/Verbose.kt
package java

object Verbose {
    fun tryCount(l: ArrayList<Int>, n: Int): Int {
        if (l.size == 0) {
            return Int.MAX_VALUE
        }

        var count = 0

        var i = 0
        while (i + 1 < l.size) {
            if (l[i] + 1 != l[i + 1]) {
                count++
            }
            i++
        }
        if (l[0] != 0) {
            count++
        }

        if (l[l.size - 1] != n - 1) {
            count++
        }

        return count
    }
}

File: datasets/JPlag-Java/to-typescript-convertai/Verbose.ts
import { ArrayList } from 'typescript-collections';

export class Verbose {
    static tryCount(l: ArrayList<number>, n: number): number {
        if (l.size() === 0) {
            return Number.MAX_SAFE_INTEGER;
        }

        let count = 0;

        for (let i = 0; i + 1 < l.size(); i++) {
            if (l.get(i) + 1 !== l.get(i + 1)) {
                count++;
            }
        }
        if (l.get(0) !== 0) {
            count++;
        }

        if (l.get(l.size() - 1) !== n - 1) {
            count++;
        }

        return count;
    }
}

